src/libs/actions/Modal.ts

```diff
@@ -1,30 +1,38 @@
 import Onyx from 'react-native-onyx';
 import ONYXKEYS from '@src/ONYXKEYS';
 
-let closeModal: (isNavigating: boolean) => void;
+const closeModals: Array<(isNavigating: boolean) => void> = [];
+
 let onModalClose: null | (() => void);
 
+/**
+ * Remove onClose function in list
+ */
+function removeCloseModal(onClose: () => void) {
+    const index = closeModals.findIndex((o) => o === onClose);
+    if (index === -1) {
+        return;
+    }
+    closeModals.splice(index, 1);
+}
+
 /**
  * Allows other parts of the app to call modal close function
  */
 function setCloseModal(onClose: () => void) {
-    closeModal = onClose;
+    closeModals.push(onClose);
```

Comment:
We should check if the function reference already exists to prevent duplicates.
```suggestion
     if (!closeModals.includes(onClose)) {
        closeModals.push(onClose);
     }
```